internal/backends/android-activity/lib.rs

```diff
@@ -340,6 +360,37 @@ impl AndroidWindowAdapter {
                     MotionAction::HoverEnter | MotionAction::HoverExit => InputStatus::Unhandled,
                     _ => InputStatus::Unhandled,
                 },
+                InputEvent::TextEvent(state) => {
+                    let runtime_window = i_slint_core::window::WindowInner::from_pub(&self.window);
+                    // remove the pre_edit
+                    let event = if let Some(r) = state.compose_region {
+                        i_slint_core::input::KeyEvent {
+                            event_type: i_slint_core::input::KeyEventType::UpdateComposition,
+                            text: i_slint_core::format!(
+                                "{}{}",
+                                &state.text[..r.start],
+                                &state.text[r.end..]
+                            ),
+                            preedit_text: state.text[r.start..r.end].into(),
+                            preedit_selection: Some(r.start as i32..r.end as i32),
+                            replacement_range: Some(i32::MIN..i32::MAX),
+                            cursor_position: Some(state.selection.end as _),
+                            anchor_position: Some(state.selection.start as _),
+                            ..Default::default()
+                        }
+                    } else {
+                        i_slint_core::input::KeyEvent {
+                            event_type: i_slint_core::input::KeyEventType::CommitComposition,
+                            text: state.text.as_str().into(),
+                            replacement_range: Some(i32::MIN..i32::MAX),
```

Comment:
In a rich-text environment this approach will be a problem. Ideally we'd just commit the composition and not replace the entire text, as the latter would delete any previously applied formatting.

Reply:
I agree, this is not something we need to deal with right now.